app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -350,13 +357,54 @@ def populate_hardware_hash_with_series_attributes(hardware_hash, instance, serie
         hardware_hash[:memory_mb]       = series[:memory] / 1.megabyte
         hardware_hash[:disk_capacity]   = series[:root_disk_size] + series[:swap_disk_size]
 
-        os_disk = instance.properties.storage_profile.os_disk
-        sz      = series[:root_disk_size]
-        vhd_loc = os_disk.try(:vhd).try(:uri) || os_disk.try(:managed_disk).try(:id)
+        add_azure_instance_disk(hardware_hash[:disks], instance)
+
+        # No data available on swap disk? Called temp or resource disk.
+      end
+
+      def add_azure_instance_disk(array, instance, os_disk = nil)
+        os_disk ||= instance.properties.storage_profile.os_disk
```

Comment:
Why did you insert the string `azure` into the method name? Its implied that this applies to Azure.

Reply:
I can rename it. I didn't want to use `add_instance_disk` since it's inherited, but I guess there's no harm in just redefining it.

Reply:
Yea you could override it.